astropy/utils/console.py

```diff
@@ -49,42 +50,33 @@ class _IPython:
 
     @classproperty
     def get_ipython(cls):
-        try:
+        if HAS_IPYTHON:
             from IPython import get_ipython
-        except ImportError:
-            pass
-        return get_ipython
+
+            return get_ipython
+        raise ModuleNotFoundError("IPython is not installed")
 
     @classproperty
     def OutStream(cls):
         if not hasattr(cls, "_OutStream"):
-            cls._OutStream = None
-            try:
-                cls.get_ipython()
-            except NameError:
-                return None
-
-            try:
+            if HAS_IPYTHON:
                 from ipykernel.iostream import OutStream
```

Comment:
`IPython.zmq` was removed in 2013 in https://github.com/ipython/ipython/pull/4121 and this functionality was moved to `IPython.kernel`, which was itself deprecated in IPython 4.0.0 and removed in 8.0.0, see https://github.com/ipython/ipython/pull/13386